### PR TITLE
Rename file term to caller

### LINKED ISSUE #432: log: default StdlibAdapter file key should be "caller"
Not "file". To match the naming of DefaultCaller/UTC.

Open to a PR?

### PR DESCRIPTION
**Pull request by willfaught**
WIP

Fixes #432.

### CODE REVIEW

File: log/stdlib.go [hunk 1]
```diff
@@ -39,9 +39,9 @@ func TimestampKey(key string) StdlibAdapterOption {
 	return func(a *StdlibAdapter) { a.timestampKey = key }
 }
 
-// FileKey sets the key for the file and line field. By default, it's "file".
-func FileKey(key string) StdlibAdapterOption {
-	return func(a *StdlibAdapter) { a.fileKey = key }
+// CallerKey sets the key for the file and line field. By default, it's "caller".
+func CallerKey(key string) StdlibAdapterOption {
```
Comment (peterbourgon) [inline log/stdlib.go]: To preserve backwards compatibility, could we leave the name of the option as `FileKey`?

File: log/stdlib.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ func (w StdlibWriter) Write(p []byte) (int, error) {
 type StdlibAdapter struct {
 	Logger
 	timestampKey string
-	fileKey      string
+	callerKey    string
```
Comment (peterbourgon) [inline log/stdlib.go]: Likewise leave this as `fileKey` with a default value of `"caller"`

File: log/stdlib.go [hunk 3]
```diff
@@ -93,14 +93,14 @@ func (a StdlibAdapter) Write(p []byte) (int, error) {
 }
 
 const (
-	logRegexpDate = `(?P<date>[0-9]{4}/[0-9]{2}/[0-9]{2})?[ ]?`
-	logRegexpTime = `(?P<time>[0-9]{2}:[0-9]{2}:[0-9]{2}(\.[0-9]+)?)?[ ]?`
-	logRegexpFile = `(?P<file>.+?:[0-9]+)?`
-	logRegexpMsg  = `(: )?(?P<msg>.*)`
+	logRegexpDate   = `(?P<date>[0-9]{4}/[0-9]{2}/[0-9]{2})?[ ]?`
+	logRegexpTime   = `(?P<time>[0-9]{2}:[0-9]{2}:[0-9]{2}(\.[0-9]+)?)?[ ]?`
+	logRegexpCaller = `(?P<caller>.+?:[0-9]+)?`
```
Comment (peterbourgon) [inline log/stdlib.go]: And likewise leave this as `logRegexpFile`. Essentially just changing the default constant.

**Comment by peterbourgon** [comment]
You'll need to update the tests, too.

**Comment by willfaught** [comment]
Will do

**Comment by willfaught** [comment]
I was hoping I could get by with editing the one file on GitHub. Will fix.

**Pull request closed without merging**